File: ui/components/center_pane.go
package components

import (
	"fmt"
	"strings"
	"time"

	tea "github.com/charmbracelet/bubbletea"
)

// FeedItem represents a feed item
type FeedItem struct {
	ID      int
	Time    time.Time
	Source  string
	Content string
	Status  string
	Read    bool
}

// ChatMessage represents a chat message
type ChatMessage struct {
	Sender  string
	Content string
	Time    time.Time
}

// CenterPane represents the center pane with agent chat and feed
type CenterPane struct {
	width        int
	height       int
	cursor       int
	feedItems    []FeedItem
	chatMessages []ChatMessage
	inputBuffer  string
}

// NewCenterPane creates a new center pane
func NewCenterPane() *CenterPane {
	return &CenterPane{
		feedItems: []FeedItem{
			{ID: 1, Time: time.Now(), Source: "📧 Email", Content: "Meeting reminder: Team sync at 10:00", Status: "⏰ Pending", Read: false},
			{ID: 2, Time: time.Now(), Source: "💰 Finance", Content: "Gold price: $2,150/oz (+2.3%)", Status: "📈 Alert", Read: false},
			{ID: 3, Time: time.Now(), Source: "🌪️ Weather", Content: "Hurricane warning: Category 2 approaching", Status: "⚠️ Alert", Read: false},
			{ID: 4, Time: time.Now(), Source: "📅 Calendar", Content: "Team sync meeting starting now", Status: "🎯 Active", Read: true},
			{ID: 5, Time: time.Now(), Source: "📧 Email", Content: "Project update from client", Status: "📋 New", Read: false},
		},
		chatMessages: []ChatMessage{
			{Sender: "Captain", Content: "Worker agent, status report on ui.01 contract?", Time: time.Now()},
			{Sender: "Worker", Content: "Implementing three-pane terminal interface with Crush framework. Progress: 60%", Time: time.Now()},
			{Sender: "Captain", Content: "Excellent. Focus on core functionality first.", Time: time.Now()},
			{Sender: "Worker", Content: "Acknowledged. Will complete layout and basic navigation.", Time: time.Now()},
		},
	}
}

// Init initializes the center pane
func (cp *CenterPane) Init() tea.Cmd {
	return nil
}

// Update handles updates for the center pane
func (cp *CenterPane) Update(msg tea.Msg) (*CenterPane, tea.Cmd) {
	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch msg.String() {
		case "up":
			if cp.cursor > 0 {
				cp.cursor--
			}
		case "down":
			if cp.cursor < len(cp.feedItems)-1 {
				cp.cursor++
			}
		case "enter":
			cp.markAsRead(cp.cursor)
		case "d":
			cp.markAsDone(cp.cursor)
		case "backspace":
			if len(cp.inputBuffer) > 0 {
				cp.inputBuffer = cp.inputBuffer[:len(cp.inputBuffer)-1]
			}
		default:
			if len(msg.String()) == 1 {
				cp.inputBuffer += msg.String()
			}
		}
	}
	return cp, nil
}

// View renders the center pane
func (cp *CenterPane) View() string {
	feedSection := cp.renderFeedItems()
	chatSection := cp.renderChatInterface()

	content := fmt.Sprintf("%s\n\n%s", feedSection, chatSection)

	title := "💬 Agent Chat & Feed"
	subtitle := "Press Enter to interact"

	return CenterPaneStyle.Render(fmt.Sprintf("%s\n%s\n\n%s", title, subtitle, content))
}

// SetSize sets the pane size
func (cp *CenterPane) SetSize(width, height int) {
	cp.width = width
	cp.height = height
}

// markAsRead marks a feed item as read
func (cp *CenterPane) markAsRead(index int) {
	if index >= 0 && index < len(cp.feedItems) {
		cp.feedItems[index].Read = true
	}
}

// markAsDone marks a feed item as done
func (cp *CenterPane) markAsDone(index int) {
	if index >= 0 && index < len(cp.feedItems) {
		cp.feedItems[index].Status = "✅ Done"
	}
}

// renderFeedItems renders the feed items section
func (cp *CenterPane) renderFeedItems() string {
	var lines []string
	lines = append(lines, "📰 Unified Feed")
	lines = append(lines, strings.Repeat("=", 30))
	lines = append(lines, "Time    Source    Content                    Status")
	lines = append(lines, strings.Repeat("-", 30))

	for i, item := range cp.feedItems {
		timeStr := item.Time.Format("15:04")
		line := fmt.Sprintf("%s  %s  %s  %s", timeStr, item.Source, item.Content, item.Status)

		if i == cp.cursor {
			line = CursorStyle.Render("> " + line)
		} else {
			line = "  " + line
		}

		if !item.Read {
			line = UnreadStyle.Render(line)
		}

		lines = append(lines, line)
	}

	return strings.Join(lines, "\n")
}

// renderChatInterface renders the chat interface section
func (cp *CenterPane) renderChatInterface() string {
	var lines []string
	lines = append(lines, "🤖 Agent Chat")
	lines = append(lines, strings.Repeat("=", 40))

	for _, msg := range cp.chatMessages {
		line := fmt.Sprintf("%s: %s", msg.Sender, msg.Content)
		lines = append(lines, line)
	}

	lines = append(lines, strings.Repeat("=", 40))
	lines = append(lines, fmt.Sprintf("Type your message: %s", cp.inputBuffer))

	return strings.Join(lines, "\n")
}

File: ui/components/center_pane.py
"""
Center Pane - Agent Chat Interface

Displays agent interactions and feed items with action capabilities.
"""

from typing import List, Dict, Any, Optional
from datetime import datetime
from rich.panel import Panel
from rich.text import Text
from rich.table import Table
from rich.columns import Columns
from rich.console import Group


class CenterPane:
    """Center pane showing agent chat interface and feed items."""
    
    def __init__(self, feed_connector):
        self.feed_connector = feed_connector
        self.messages = []
        self.selected_item = None
        self.input_buffer = ""
        
        # Initialize with some sample data
        self._load_sample_data()
        
    def render(self) -> Panel:
        """Render the center pane content."""
        content = Group(
            self._render_feed_items(),
            self._render_chat_interface()
        )
        
        return Panel(
            content,
            title="💬 Agent Chat & Feed",
            border_style="green",
            subtitle="Press Enter to interact"
        )
    
    def _render_feed_items(self) -> Table:
        """Render the feed items table."""
        table = Table(title="📰 Unified Feed", show_header=True)
        table.add_column("Time", style="cyan", no_wrap=True)
        table.add_column("Source", style="magenta")
        table.add_column("Content", style="white")
        table.add_column("Status", style="yellow")
        
        # Sample feed items
        feed_items = [
            ("09:15", "📧 Email", "Meeting reminder: Team sync at 10:00", "⏰ Pending"),
            ("09:30", "💰 Finance", "Gold price: $2,150/oz (+2.3%)", "📈 Alert"),
            ("09:45", "🌪️ Weather", "Hurricane warning: Category 2 approaching", "⚠️ Alert"),
            ("10:00", "📅 Calendar", "Team sync meeting starting now", "🎯 Active"),
            ("10:15", "📧 Email", "Project update from client", "📋 New"),
        ]
        
        for time, source, content, status in feed_items:
            table.add_row(time, source, content, status)
            
        return table
    
    def _render_chat_interface(self) -> Text:
        """Render the agent chat interface."""
        chat_text = Text()
        chat_text.append("🤖 Agent Chat\n", style="bold green")
        chat_text.append("=" * 40 + "\n", style="blue")
        
        # Sample chat messages
        chat_messages = [
            ("Captain", "Worker agent, status report on ui.01 contract?"),
            ("Worker", "Implementing three-pane terminal interface with Rich library. Progress: 60%"),
            ("Captain", "Excellent. Focus on core functionality first."),
            ("Worker", "Acknowledged. Will complete layout and basic navigation."),
        ]
        
        for sender, message in chat_messages:
            chat_text.append(f"{sender}: ", style="bold cyan")
            chat_text.append(f"{message}\n", style="white")
        
        chat_text.append("\n" + "=" * 40 + "\n", style="blue")
        chat_text.append("Type your message: ", style="bold yellow")
        chat_text.append(self.input_buffer, style="white")
        
        return chat_text
    
    def _load_sample_data(self):
        """Load sample data for demonstration."""
        self.messages = [
            {
                "timestamp": datetime.now(),
                "sender": "Captain",
                "content": "Worker agent, begin implementation of ui.01 contract.",
                "type": "command"
            },
            {
                "timestamp": datetime.now(),
                "sender": "Worker",
                "content": "Acknowledged. Starting three-pane terminal interface implementation.",
                "type": "response"
            }
        ]
    
    def add_message(self, sender: str, content: str, msg_type: str = "message"):
        """Add a new message to the chat."""
        self.messages.append({
            "timestamp": datetime.now(),
            "sender": sender,
            "content": content,
            "type": msg_type
        })
    
    def select_feed_item(self, index: int):
        """Select a feed item for action."""
        self.selected_item = index
    
    def mark_item_done(self, index: int):
        """Mark a feed item as done."""
        if 0 <= index < len(self.messages):
            # In a real implementation, this would update the feed connector
            pass
    
    def delete_item(self, index: int):
        """Delete a feed item."""
        if 0 <= index < len(self.messages):
            # In a real implementation, this would update the feed connector
            pass
    
    def set_input_buffer(self, text: str):
        """Set the input buffer text."""
        self.input_buffer = text
